File: cosmobusiness/src/java/it/csi/cosmo/cosmobusiness/business/service/AttivazioneSistemaEsternoService.java
/*
 * Copyright CSI-Piemonte - 2023
 * SPDX-License-Identifier: GPL-3.0-or-later
 */

package it.csi.cosmo.cosmobusiness.business.service;

import it.csi.cosmo.cosmobusiness.dto.rest.GetPayloadAttivazioneSistemaEsternoResponse;
import it.csi.cosmo.cosmobusiness.dto.rest.RiferimentoOperazioneAsincrona;

public interface AttivazioneSistemaEsternoService {

  /**
   * @param idPratica
   * @param idAttivita
   * @return
   */
  RiferimentoOperazioneAsincrona inviaAttivazioneSistemaEsterno(Long idPratica, Long idAttivita);

  /**
   * @param idPratica
   * @param idAttivita
   * @return
   */
  GetPayloadAttivazioneSistemaEsternoResponse getPayloadAttivazioneSistemaEsterno(Long idPratica,
      Long idAttivita);

}

File: cosmowcl/src/app/shared/services/attivazione-sistema-esterno.service.ts
/*
 * Copyright CSI-Piemonte - 2023
 * SPDX-License-Identifier: GPL-3.0-or-later
 */
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { OperazioneAsincrona } from '../models/api/cosmobusiness/operazioneAsincrona';
import { ApiUrls } from '../utilities/apiurls';
import { Utils } from '../utilities/utilities';

@Injectable({
  providedIn: 'root'
})
export class AttivazioneSistemaEsternoService {

  constructor(
    private http: HttpClient
  ) {
    // NOP
  }

  getPayload(idPratica: number, idAttivita: number) {
    // tslint:disable-next-line:max-line-length
    return this.http.get(ApiUrls.GET_PAYLOAD_ATTIVAZIONE_SISTEMA_ESTERNO
        .replace('{idPratica}', idPratica.toString()).replace('{idAttivita}', idAttivita.toString()));
  }

  setAttivazione(idPratica: number, idAttivita: number): Observable<OperazioneAsincrona>{
    return this.http.post<OperazioneAsincrona>(ApiUrls.POST_ATTIVAZIONE_SISTEMA_ESTERNO
      .replace('{idPratica}', idPratica.toString())
        .replace('{idAttivita}', idAttivita.toString()), {});
  }

}
